### PR TITLE
feat(ingest): Allow csv-enricher to update more types

### PR DESCRIPTION
**Pull request by xiphl**
Currently, csv-enricher can only update datasets. My organisation would like to expand csv-enricher to allow container descriptions to be updated as well, as they are generated together with datasets during ingestion.
With this PR, more entity types can update the description via csv-enricher. However, glossary terms and node groups are not included in this PR, as they lack the "editableProperties" aspect.

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/csv_enricher.py [hunk 1]
```diff
@@ -264,28 +275,73 @@ def get_resource_description_work_unit(
...
+
+        # as not all classes have the same attributes
+        additionalAttributesClass = (
+            EditableDatasetPropertiesClass,
+            EditableChartPropertiesClass,
+            EditableDashboardPropertiesClass,
+            EditableDataJobPropertiesClass,
+            EditableDataFlowPropertiesClass,
+            EditableNotebookPropertiesClass,
+        )
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/csv_enricher.py:341]: we can do this without hardcoding by using a hasattr check

```py
from datahub.metadata.schema_classes import (
    EditableDatasetPropertiesClass,
    EditableMLModelGroupPropertiesClass,
)

print(hasattr(EditableDatasetPropertiesClass, "created"))
print(hasattr(EditableMLModelGroupPropertiesClass, "created"))
```

File: metadata-ingestion/src/datahub/ingestion/source/csv_enricher.py [hunk 2]
```diff
@@ -264,28 +275,73 @@ def get_resource_description_work_unit(
         # Check if there is a description to add. If not, return None.
         if not description:
             return None
-
         # If the description is empty, return None.
         if len(description) <= 0:
             return None
-
-        current_editable_properties: Optional[EditableDatasetPropertiesClass] = None
+        entityType = entity_urn.replace("urn:li:", "", 1).split(":")[0]
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/csv_enricher.py:281]: we have a guess_entity_type() helper method that we can use instead

**Review by hsheth2 — COMMENTED** [review]
This is awesome! Thanks @xiphl 

Had a few small comments and the lint is failing, but should be good to merge soon

File: metadata-ingestion/src/datahub/ingestion/source/csv_enricher.py (hunk 1 above)
Comment (xiphl) [reply metadata-ingestion/src/datahub/ingestion/source/csv_enricher.py]: mypy really hates this one trick! :)

**Review by hsheth2 — APPROVED** [review]

**Pull request merged**